mlflow/pyfunc/__init__.py

```diff
@@ -3035,6 +3040,23 @@ def predict(self, context, model_input: List[str], params=None) -> List[str]:
         auth_policy: {{ auth_policy }}
         kwargs: Extra keyword arguments.
     """
+    if not isinstance(python_model, (Path, str)):
+        if not MLFLOW_ALLOW_UNSAFE_PICKLE_DESERIALIZATION.get():
+            raise MlflowException(
+                "Unsafe pickler deserialization for custom python model is disallowed. "
+                "Please set 'python_model' parameter to a file path to the PythonModel which "
+                "defines the model from code artifact to avoid using unsafe pickler, or set "
+                "environment variable 'MLFLOW_ALLOW_UNSAFE_PICKLE_DESERIALIZATION' to 'true' "
+                "to allow unsafe pickler."
+            )
+        warnings.warn(
+            "Saving custom python model by unsafe pickler is deprecated, and will be disabled "
+            "by default in future MLflow versions. Saving python model as the 'model from code' "
+            "artifact is the recommended way.",
```

Comment:
The current message implicitly assumes the user knows what the model from code artifact is. That might not be the case.